Packages/com.unity.inputsystem/InputSystem/Controls/InputControlExtensions.cs

```diff
@@ -1896,6 +1897,26 @@ public DeviceBuilder WithStateOffsetToControlIndexMap(uint[] map)
                 return this;
             }
 
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public unsafe DeviceBuilder WithControlTree(byte[] controlTreeNodes, ushort[] controlTreeIndicies)
+            {
+                var sizeOfNode = UnsafeUtility.SizeOf<InputDevice.ControlBitRangeNode>();
+                var numNodes = controlTreeNodes.Length / sizeOfNode;
+                device.m_ControlTreeNodes = new InputDevice.ControlBitRangeNode[numNodes];
+                fixed (byte* nodePtr = controlTreeNodes)
+                {
+	                for (var i = 0; i < numNodes; i++)
+	                {
+		                device.m_ControlTreeNodes[i] = *(InputDevice.ControlBitRangeNode*)(nodePtr + i * sizeOfNode);
+		                // device.m_ControlTreeNodes[i] = MemoryMarshal.Read<InputDevice.ControlBitRangeNode>(
```

Comment:
Intentionally uncommented code or should it be removed?

Reply:
Should have been removed.